tracy-client/src/lib.rs

```diff
@@ -34,18 +37,33 @@ macro_rules! span {
         $crate::span!($name, 62)
     };
     ($name:expr, $callstack_depth:expr) => {{
+        use std::ffi::CString;
+        use $crate::once_cell::sync::Lazy;
+
         struct S;
-        let func_name = std::any::type_name::<S>();
-        $crate::Span::new(
-            $name,
-            &func_name[..func_name.len() - 3],
-            file!(),
-            line!(),
-            $callstack_depth,
-        )
+        static FUNC_NAME: Lazy<CString> = Lazy::new(|| {
+            let func_name = std::any::type_name::<S>();
+            CString::new(&func_name[..func_name.len() - 3]).unwrap()
+        });
+        static SRC_LOC: Lazy<$crate::SrcLoc> = Lazy::new(|| {
+            $crate::SrcLoc($crate::sys::___tracy_source_location_data {
+                name: concat!($name, "\0").as_ptr().cast(),
+                function: FUNC_NAME.as_ptr(),
```

Comment:
We can probably avoid the two `Lazy`s per span site, right? You could probably just create the `CString` in this same `SRC_LOC` `Lazy`, grab the pointer and `std::mem::forget` the `CString` itself.

(And if the two `Lazy`s do in fact serve some purpose, a comment would be nice.)

Reply:
Refactored to use a single `Lazy`.